code/modules/clothing/chameleon.dm

```diff
@@ -580,6 +580,40 @@
 	var/datum/action/item_action/chameleon/drone/randomise/randomise_action = new(src)
 	randomise_action.UpdateButtonIcon()
 
+/datum/action/item_action/chameleon/tongue_change
+	name = "Tongue Change"
+	icon_icon = 'icons/obj/surgery.dmi'
+	button_icon_state = "tonguebone"
+	var/list/tongue_list = list()
+
+/datum/action/item_action/chameleon/tongue_change/proc/generate_tongue_list()
+	var/obj/item/found_item
+	var/tongue_name
+	var/static/list/predefined_tongues = typesof(/obj/item/organ/tongue)
+	for(var/found_var in predefined_tongues)
+		found_item = found_var
+		if((initial(found_item.item_flags) & ABSTRACT) || !initial(found_item.icon_state))
+			continue
+		tongue_name = "[initial(found_item.name)] ([initial(found_item.icon_state)])"
+		tongue_list[tongue_name] = found_item
+
+/datum/action/item_action/chameleon/tongue_change/Trigger()
+	if(!IsAvailable() || !isitem(target))
+		return FALSE
+	var/obj/item/clothing/mask/target_mask = target
+	var/obj/item/organ/tongue/picked_tongue
+	var/picked_name
+	var/list/sorted_list = sort_list(tongue_list)
```

Comment:
Sort it on generation instead of inside here

Reply:
Done